openlibrary/plugins/importapi/import_validator.py

```diff
@@ -1,24 +1,185 @@
-from typing import Annotated, Any, TypeVar
+# mostly generated by datamodel-codegen:
+#   filename:  import.schema.json
+#   timestamp: 2023-07-24T12:05:23+00:00
+
+from __future__ import annotations
+
+from enum import Enum
+from typing import Annotated, Any, Optional, TypeVar
 
 from annotated_types import MinLen
-from pydantic import BaseModel, ValidationError
+from pydantic import (
+    AnyHttpUrl,
+    BaseModel,
+    ConfigDict,
+    Field,
+    PositiveInt,
+    RootModel,
+    StringConstraints,
+    ValidationError,
+)
 
 T = TypeVar("T")
 
 NonEmptyList = Annotated[list[T], MinLen(1)]
-NonEmptyStr = Annotated[str, MinLen(1)]
+NonEmptyStr = Annotated[str, StringConstraints(strip_whitespace=True, min_length=1)]
+
+
+class PublishCountry(RootModel):
+    root: Annotated[str, StringConstraints(pattern=r'^[a-z]{2,3}$')] = Field(
+        ...,
+        description='The MARC21 country code. See https://www.loc.gov/marc/countries/cou_home.html',
+        examples=['enk', 'gw', 'flu'],
+    )
+
+
+class LanguageCode(RootModel):
+    root: Annotated[str, StringConstraints(pattern=r'^[a-z]{3}$')] = Field(
+        ...,
+        description='The MARC21 language code. See https://www.loc.gov/marc/languages/language_code.html',
+        examples=['eng', 'fre', 'ger', 'tha'],
+    )
+
+
+class Isbn10(RootModel):
+    root: Annotated[str, StringConstraints(pattern=r'^([0-9][- ]*){9}[0-9X]$')]
```

Comment:
Both the ISBN 10 and 13 regexes could be changed to only match one-or-zero hyphens or spaces (use `?` instead of `*`):

`^([0-9][- ]?){9}[0-9X]$`

Otherwise things like `978----------------0123456789 `  will be accepted.